crates/rethnet_eth/src/remote.rs

```diff
@@ -39,6 +39,53 @@ impl From<u64> for U64 {
     }
 }
 
+/// to represent the Object { blockHash, requireCanonical } in EIP-1898
+#[derive(Clone, Debug, PartialEq, serde::Deserialize, serde::Serialize)]
+pub struct Eip1898BlockHash {
+    /// the block hash
+    pub block_hash: B256,
+    /// whether the server should additionally raise a JSON-RPC error if the block is not in the
+    /// canonical chain
+    pub require_canonical: Option<bool>,
+}
+
+/// to represent the Object { blockNumber } in EIP-1898
+#[derive(Clone, Debug, PartialEq, serde::Deserialize, serde::Serialize)]
+pub struct Eip1898BlockNumber {
+    /// the block number
+    pub block_number: U256,
+}
+
+///
+#[derive(Clone, Debug, PartialEq, serde::Deserialize, serde::Serialize)]
+#[serde(untagged)]
+pub enum Eip1898BlockSpec {
+    /// to represent the Object { blockHash, requireCanonical } in EIP-1898
+    Hash(Eip1898BlockHash),
+    /// to represent the Object { blockNumber } in EIP-1898
+    Number(Eip1898BlockNumber),
```

Comment:
Why not inline the arguments?

To show the usage difference in Rust:

```
match spec {
-  Eip1898BlockSpec::Number(Eip1898BlockNumber { block_number }) => ...,
+ Eip1898BlockSpec::Number { block_number } => ...,
}
```

or

```
match spec {
-  Eip1898BlockSpec::Number(block_number) => block_number.block_number,
+ Eip1898BlockSpec::Number{ block_number => block_number,
}
```